DAQ/src/ArtBinaryPacketsFromDigis_module.cc

```diff
@@ -0,0 +1,1113 @@
+    void   fillTrackerDataPacket(const StrawDigi& SD, TrackerDataPacket& TrkData);
+
+    void   fillTrackerHeaderDataPacket(const StrawDigi& SD, DataBlockHeader& HeaderData, uint64_t& EventNum);
+
+    void   fillEmptyTrackerDataPacket(TrackerDataPacket& TrkData);
+
+    void   processTrackerData(art::Event& evt, uint64_t& eventNum,
+			      tracker_data_block_list_t &trackerData);
+
+
+    void   fillTrackerDMABlocks(raw_data_list_t& dataStream, tracker_data_block_list_t const& trackerData);
+
+    void   fillTrackerDataStream(raw_data_list_t& dataStream, tracker_data_block_t const& trkData);
+
+    void   printTrackerData(TrackerDataPacket const& curDataBlock);
+
+    //--------------------------------------------------------------------------------
+    //  methods used to handle the calorimeter data
+    //--------------------------------------------------------------------------------
+    void   fillCalorimeterDataPacket(const CaloDigi& SD, CaloDataPacket& caloData);
+    void   addCaloHitToCaloPacket(calo_data_block_t& dataBlock,
+				  CaloDataPacket& caloData);
+
+    void   fillCalorimeterHeaderDataPacket(const CaloDigi& SD, DataBlockHeader& HeaderData, uint64_t& EventNum);
+
+    void   fillEmptyCalorimeterDataPacket(CaloDataPacket& caloData);
+    void   fillHeaderByteAndPacketCounts(calo_data_block_t& caloData);
+
+    void   processCalorimeterData(art::Event& evt, uint64_t& eventNum,
+				  calo_data_block_list_t& caloDataBlocks);
+
+
+    void   fillCalorimeterDMABlocks(raw_data_list_t& dataStream, calo_data_block_list_t& caloData);
+
+    void   fillCalorimeterDataStream(raw_data_list_t& dataStream, calo_data_block_t& caloData);
+
+    void   printCalorimeterData(CaloDataPacket const& curDataBlock);
+
+    size_t waveformMaximumIndex(std::vector<adc_t>& waveform);
+
+    void flushBuffer(raw_data_list_t& buffers);
+    void closeDataBuffer(raw_data_list_t& dataStream, bool openNew = true) {     
+      uint64_t sz = dataStream.back().second;
+      if (_diagLevel > 3) {
+	std::cout<< "[closeDataBuffer] 1) sz = "<<sz<<std::endl;
+	std::cout << "[closeDataBuffer] 1a) sz = "<< dataStream.back().second << std::endl;
+      }
+      if (sz < 48) {
+	if (_diagLevel > 3) {
+	  std::cout<< "[closeDataBuffer] dataStream.back().first = "<<dataStream.back().first<<std::endl;
+	  std::cout<< "[closeDataBuffer] dataStream.back().first[sz] = "<<dataStream.back().first[sz]<<std::endl;
+	  std::cout<< "[closeDataBuffer] &dataStream.back().first[sz] = "<<&dataStream.back().first[sz]<<std::endl;
+	}
+	bzero(&dataStream.back().first[sz], 48 - sz);
+	sz = 48;
+      }
+      if (_diagLevel > 3) {
+	std::cout << "[closeDataBuffer] dataStream.size = "<< dataStream.size() << std::endl;
+	if (dataStream.size()> 0){
+	  std::cout <<  "[closeDataBuffer] dataStream.back().second = " << dataStream.back().second <<std::endl;
+	}
+	std::cout<< "[closeDataBuffer] 2) sz = "<<sz<<std::endl;
+      }
+      uint64_t ex_sz = sz - 16;
+
```

Comment:
The above has a lot of numeric literals: 2, 3, 16, 48 ...  Each should be defined as an appropriate symbolic, either constexpr or a variable coming from Proditions.